src/gui.cpp

```diff
@@ -1417,27 +1420,47 @@ moby_model* gui::model_browser::render_selection_grid(
 
 void gui::model_browser::render_preview(
 		app& a,
-		GLuint* target,
+		GLuint* target,	
 		moby_model& model,
 		std::vector<texture>& textures,
 		const gl_renderer& renderer,
 		ImVec2 preview_size,
 		view_params params) {
-	glm::vec3 eye = glm::vec3(2.f * (1.1f - params.zoom), 0, 0);
+
+	auto draw_list = ImGui::GetWindowDrawList();
+
+    glm::vec3 center_point = (model.bounding_box.max + model.bounding_box.min) * 0.5f / (float) INT16_MAX; 
 	
-	glm::mat4 view_fixed = glm::lookAt(eye, glm::vec3(0, 0, 0), glm::vec3(0, 1, 0));
+	//Figure out a rough camera distance 
+	float bounding_box_width = glm::abs(model.bounding_box.max.x - model.bounding_box.min.x),
+			bounding_box_height = glm::abs(model.bounding_box.max.y - model.bounding_box.min.y),
+			fov_y = 45.0f,
+			camera_distance;
+
+	if (bounding_box_width > bounding_box_height) {
+		float fov_x = fov_y * preview_size.x / preview_size.y;
+		//Get a radio of how wide the bounding box of the model is compared to the render window width.
+		float zoom_ratio = bounding_box_height / preview_size.x / (float) INT16_MAX * 10.0f;
+		//Figure out the 'default' camera depth to fit an object of width preview_size.x on screen.
+		float initial_depth = (preview_size.x/2) * tan(glm::radians(fov_x));
+		//Fit the camera to the model bounding box.
+		camera_distance = zoom_ratio * initial_depth;
+	} else {	//Same as above, but use height if the model is taller than wide.
+		float zoom_ratio = bounding_box_height / preview_size.y / (float) INT16_MAX * 10.0f;
+		float initial_depth = (preview_size.y/2) * tan(glm::radians(fov_y));
+		camera_distance = zoom_ratio * initial_depth;
+	}
+
+	glm::vec3 eye = glm::vec3(((camera_distance) * (1.1f - params.zoom)), 0, 0);
+
+	glm::mat4 view_fixed = glm::lookAt(eye, glm::vec3(0), glm::vec3(0, 1, 0));
 	glm::mat4 view_pitched = glm::rotate(view_fixed, params.pitch_yaw.x, glm::vec3(0, 0, 1));
 	glm::mat4 view = glm::rotate(view_pitched, params.pitch_yaw.y, glm::vec3(0, 1, 0));
-	glm::mat4 projection = glm::perspective(glm::radians(45.0f), preview_size.x / preview_size.y, 0.01f, 100.0f);
+	glm::mat4 offset_view = glm::translate(view, -center_point);
+
+	glm::mat4 projection = glm::perspective(glm::radians(fov_y), preview_size.x / preview_size.y, 0.01f, 100.0f);
 	
-	static const glm::mat4 yzx {
-		0,  0, 1, 0,
-		1,  0, 0, 0,
-		0, -1, 0, 0,
-		0,  0, 0, 1
-	};
-	glm::mat4 local_to_world = glm::translate(glm::mat4(1.f), glm::vec3(0.f, 0.f, -0.125f));
-	glm::mat4 local_to_clip = projection * view * yzx * local_to_world;
```

Comment:
These two additional matrix multiplications seemed to just offset the models incorrectly. Am I missing something important regarding them?

Reply:
They were just to get the ARMOR.WAD models looking a bit nicer. They shouldn't be necessary with this PR though.

Reply:
I didn't notice you also removed the multiplication by yzx. See my other more recent comments.